blockchain/blockchain.go

```diff
@@ -1397,6 +1415,41 @@ func (bc *BlockChain) gcCachedNodeLoop() {
 	}()
 }
 
+func (bc *BlockChain) pruneTrieNodeLoop() {
+	// ReadPruningMarks(1, limit) is very slow because it iterates over the most of MiscDB.
+	// ReadPruningMarks(start, limit) is much faster because it only iterates a small range.
+	startNum := uint64(1)
+
+	bc.wg.Add(1)
+	go func() {
+		defer bc.wg.Done()
+		for {
+			select {
+			case num := <-bc.chPrune:
```

Comment:
Do we have to get notified on every block? How about subscribing `ChainHeadEvent` which is a latest block event after inserting blocks?

Reply:
Well, it  doesn't really matter if it is not possible that `bc.chPrune` is full. And it won't be full since the logics are only related to db read/write only. Let me resolve this.

Reply:
FYI, `bc.chPrune` is notified only when `isCommitTrieRequired(bc, block.NumberU64())` - in every other 128 blocks.

Reply:
Oh, okay. Thanks.